tests/generate_status_reference_data.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import argparse
+import os
+
+import signac
+from define_status_test_project import _TestProject
+
+import flow
+import flow.environments
+from flow.scheduling.fake_scheduler import FakeScheduler
+
+PROJECT_NAME = "StatusTest"
+STATUS_OPTIONS_PROJECT_NAME = "StatusOptionsData"
+ARCHIVE_PATH = os.path.normpath(
+    os.path.join(os.path.dirname(__file__), "./status_reference_data.tar.gz")
+)
+
+
+def init(project):
+    """Initialize the data space for the given project."""
+
+    for a in range(2, 4):
+        for b in range(2):
+            project.open_job(dict(a=a, b=b)).init()
+            project.open_job(dict(a=dict(a=a), b=b)).init()
+
+
+def init_status_options(project):
+
+    options = [
+        {},  # default options
+        {"overview": False},  # --no-overview
+        {"overview_max_lines": 2},  # --overview-max-lines 2
+        {"detailed": True},  # -d, --detailed
+        {"parameters": ["a"]},  # -p a, --parameters a
+        {"param_max_width": 1},  # --param-max-width 1
+        {"expand": True},  # -e, --expand
+        {"all_ops": True},  # -a, --all-operations
+        {"only_incomplete": True},  # --only-incomplete-operations
+        {"dump_json": True},  # --json
+        {"unroll": False},  # --stack
+        {"compact": True},  # -1, --one-line
+        {"pretty": True},  # --pretty
+        #    {"profile": True},  # No CLI flag
```

Comment:
Update this comment to indicate why you skipped `profile`. Maybe just indicate there is a problem of some kind.

Reply:
When I set the option `profile` to `True`, I got the following error:
```
Traceback (most recent call last):
  File "/Users/zshiqi/code/signac-flow/tests/generate_status_reference_data.py", line 103, in <module>
    main(parser.parse_args())
  File "/Users/zshiqi/code/signac-flow/tests/generate_status_reference_data.py", line 88, in main
    fp.print_status(**kwargs, file=status_file)
  File "/Users/zshiqi/code/signac-flow/flow/project.py", line 2684, in print_status
    if re.match(profile, module_fn):
  File "/Users/zshiqi/miniconda3/envs/signac-flow-dev/lib/python3.9/re.py", line 191, in match
    return _compile(pattern, flags).match(string)
  File "/Users/zshiqi/miniconda3/envs/signac-flow-dev/lib/python3.9/re.py", line 303, in _compile
    raise TypeError("first argument must be string or compiled pattern")
TypeError: first argument must be string or compiled pattern
```

I discussed this with @atravitz during the office hour, and we think this might be a bug in the original code.

Reply:
This does seem to be a bug. Rerequest a review from Bradley, and we should create an issue for this unless you want to fix the bug in this PR. @Charlottez112

Reply:
> This does seem to be a bug. Rerequest a review from Bradley, and we should create an issue for this unless you want to fix the bug in this PR. @Charlottez112

I'd like to help fix this but I don't really know what `re.match` is trying to match. I'm not sure what `prof.merged_file_dict` is. I'll create the issue after Bradley takes a look at this.